pom.xml

```diff
@@ -225,6 +275,12 @@
             <artifactId>slf4j-simple</artifactId>
             <version>1.7.21</version>
             <scope>test</scope>
+            <exclusions>
+              <exclusion>
+                <groupId>org.slf4j</groupId>
+                <artifactId>slf4j-api</artifactId>
+              </exclusion>
+            </exclusions>
```

Comment:
?

Reply:
we have many dependent packages use different version of this package so i have to exclude all of them and leave one version. 
Is there some other way?

Reply:
Here is the example:
`Dependency convergence error for org.slf4j:slf4j-api:1.7.21 paths to dependency are:
+-net.snowflake:snowflake-ingest-sdk:1.1.0
  +-org.slf4j:slf4j-api:1.7.21
and
+-net.snowflake:snowflake-ingest-sdk:1.1.0
  +-org.slf4j:slf4j-simple:1.7.21
    +-org.slf4j:slf4j-api:1.7.21
and
+-net.snowflake:snowflake-ingest-sdk:1.1.0
  +-org.apache.parquet:parquet-common:1.12.3
    +-org.slf4j:slf4j-api:1.7.22
and
+-net.snowflake:snowflake-ingest-sdk:1.1.0
  +-org.apache.arrow:arrow-memory-core:10.0.0
    +-org.slf4j:slf4j-api:1.7.25
and
+-net.snowflake:snowflake-ingest-sdk:1.1.0
  +-org.apache.parquet:parquet-column:1.12.3
    +-org.apache.parquet:parquet-encoding:1.12.3
      +-org.slf4j:slf4j-api:1.7.22
and
+-net.snowflake:snowflake-ingest-sdk:1.1.0
  +-org.apache.parquet:parquet-column:1.12.3
    +-org.slf4j:slf4j-api:1.7.22
`

Reply:
Need to add a dependencyManagement section to set versions of transitive dependencies:
https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html

Example: https://github.com/snowflakedb/snowflake-kafka-connector/blob/0db1a86423e5d3858ba32a70420326e34caa433b/pom.xml#L79-L99

Reply:
I see, thanks, I will need to read the doc first too:)